godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/Basis.kt

```diff
@@ -9,64 +9,27 @@ import kotlinx.cinterop.*
 import kotlin.math.*
 
 
-class Basis(
-    p_x: Vector3,
-    p_y: Vector3,
-    p_z: Vector3
-) : CoreType {
+class Basis() : CoreType {
 
     @PublishedApi
-    internal var _x = Vector3(p_x)
+    internal var _x = Vector3()
 
     @PublishedApi
-    internal var _y = Vector3(p_y)
+    internal var _y = Vector3()
 
     @PublishedApi
-    internal var _z = Vector3(p_z)
+    internal var _z = Vector3()
 
-
-    //PROPERTIES
-    /** Return a copy of the x Vector3
-     * Warning: Writing x.x = 2 will only modify a copy, not the actual object.
-     * To modify it, use x().
-     * */
-    var x
-        get() = Vector3(_x)
-        set(value) {
-            _x = Vector3(value)
-        }
-
-    inline fun <T> x(block: Vector3.() -> T): T {
-        return _x.block()
-    }
-
-    /** Return a copy of the y Vector3
-     * Warning: Writing y.x = 2 will only modify a copy, not the actual object.
-     * To modify it, use y().
-     * */
-    var y
-        get() = Vector3(_y)
-        set(value) {
-            _y = Vector3(value)
-        }
-
-
-    inline fun <T> y(block: Vector3.() -> T): T {
-        return _y.block()
-    }
-
-    /** Return a copy of the z Vector3
-     * Warning: Writing z.x = 2 will only modify a copy, not the actual object.
-     * To modify it, use z().
-     * */
-    var z
-        get() = Vector3(_z)
-        set(value) {
-            _z = Vector3(value)
-        }
-
-    inline fun <T> z(block: Vector3.() -> T): T {
-        return _z.block()
+    init {
+        _x[0] = 1.0
```

Comment:
We don't need to use array access operator that uses switch. We already know which one we want.

Reply:
Heretic

Reply:
I'll sneak in more switches @CedNaru 😂 . Kidding aside, yep I'll fix that.